src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/TestFiles/RazorSourceGeneratorComponentTests/ScriptTag_WithVariable/8/Views_Home_Index.html

```diff
@@ -0,0 +1,33 @@
+﻿<script>console.log('What&#x27;s up');</script>
+<div>console.log('What&#x27;s up');</div>
+<script>console.log('No variable');</script>
+<div>console.log('No variable');</div>
+<script>
+    console.log('What&#x27;s up');
+</script>
+<div>
+    console.log('What&#x27;s up');
+</div>
+<script>
+    console.log('No variable');
+</script>
+<div>
+    console.log('No variable');
+</div>
+Component:
+<script>console.log('What\u0027s up');</script>
+<div>console.log(&#x27;What&#x27;s up&#x27;);</div>
+<script>console.log('No variable');</script>
+<div>console.log('No variable');</div>
+<script>
+    console.log('What\u0027s up');
+</script>
+<div>
+    console.log('What&#x27;s up');
```

Comment:
Why is the value not encoded here but is encoded on the single line `div` above?

Reply:
That's an optimization in the ComponentMarkupEncodingPass. The single line `div` can use `AddContent` which is faster (sometimes, see [this comment](https://github.com/dotnet/aspnetcore/issues/52547#issuecomment-1898257149) for details), so it's preferred (and it encodes the content). This `div` here contains newlines, so we have to use `AddMarkupContent` which doesn't encode the content.